### PR TITLE
Error handling, output formatting

### PR DESCRIPTION
**Pull request by coolguru**
- Add support for —son and —verbose flags
- Error handling
- Code clean

Apologize for a huge PR!

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -56,8 +56,8 @@ type Client struct {
 // MakeClient make client
 func MakeClient(apiKey, secretKey, endpoint string) (*Client, error) {
 
-	if apiKey == content.NONE || secretKey == content.NONE {
-		return &Client{}, fmt.Errorf(content.ERROR_MISSING_API_KEY_SECRET_KEY)
+	if apiKey == "None" || secretKey == "None" || apiKey == "" || secretKey == "" {
+		return nil, NewError("Missing API key or/and Secret key. Please run 'Coreo configure' to configure them.")
```
Comment (smattes) [inline client/client.go]: Coreo should be lowercase in 'Coreo configure'

File: client/client.go [hunk 2]
```diff
@@ -56,8 +56,8 @@ type Client struct {
 // MakeClient make client
 func MakeClient(apiKey, secretKey, endpoint string) (*Client, error) {
 
-	if apiKey == content.NONE || secretKey == content.NONE {
-		return &Client{}, fmt.Errorf(content.ERROR_MISSING_API_KEY_SECRET_KEY)
+	if apiKey == "None" || secretKey == "None" || apiKey == "" || secretKey == "" {
+		return nil, NewError("Missing API key or/and Secret key. Please run 'coreo configure' to configure them.")
```
Comment (pauldallen) [inline client/client.go]: Shouldnt this be content.ERROR_MISSING_API_KEY_SECRET_KEY ??? or something similar?

File: client/cloudAccount.go [hunk 3]
```diff
@@ -28,30 +41,36 @@ func (c *Client) GetCloudAccounts(ctx context.Context, teamID string) ([]CloudAc
...
+			e = c.Do(ctx, "GET", cloudLink.Href, nil, &clouds)
+			if e != nil {
+				return nil, NewError(err.Error())
 			}
 		}
 	}
 
+	if len(clouds) == 0 {
+		if err != nil {
+			return nil, NewError(fmt.Sprintf("No cloud accounts found under team ID %s.", teamID))
```
Comment (pauldallen) [inline client/cloudAccount.go]: eternalize error message

File: client/cloudAccount.go [hunk 4]
```diff
@@ -61,12 +80,16 @@ func (c *Client) GetCloudAccountByID(ctx context.Context, teamID, cloudID string
 		}
 	}
 
+	if cloudAccount.ID == "" {
+		return nil, NewError(fmt.Sprintf("No cloud account with ID %s found under team ID %s.", cloudID, teamID))
```
Comment (pauldallen) [inline client/cloudAccount.go]: externalize error message

File: client/composite.go [hunk 5]
```diff
@@ -20,33 +34,39 @@ type Composite struct {
...
 
 	err = c.Do(ctx, "GET", compsoitesLink.Href, nil, &composites)
 	if err != nil {
-		return composites, err
+		return nil, NewError(err.Error())
+	}
+
+	if len(composites) == 0 {
+		if err != nil {
+			return nil, NewError(fmt.Sprintf("No composites found under team ID %s.", teamID))
```
Comment (pauldallen) [inline client/composite.go]: externalize error message

File: client/gitKey.go [hunk 6]
```diff
@@ -18,40 +31,46 @@ type GitKey struct {
...
+			e = c.Do(ctx, "GET", gitKeyLink.Href, nil, &gitKeys)
+			if e != nil {
+				return nil, NewError(err.Error())
 			}
 		}
 	}
 
+	if len(gitKeys) == 0 {
+		if err != nil {
+			return nil, NewError(fmt.Sprintf("No composites found under team ID %s.", teamID))
```
Comment (pauldallen) [inline client/gitKey.go]: externalize error message

File: client/plan.go [hunk 7]
```diff
@@ -41,42 +55,52 @@ type PlanConfig struct {
...
+	plans := []*Plan{}
 	err = c.Do(ctx, "GET", plansLink.Href, nil, &plans)
 	if err != nil {
-		return plans, err
+		return nil, err
+	}
+
+	if len(plans) == 0 {
+		if err != nil {
+			return nil, NewError(fmt.Sprintf("No plans found under team ID %s and composite ID %s.", teamID, compositeID))
```
Comment (pauldallen) [inline client/plan.go]: externalize error message

File: client/plan.go [hunk 8]
```diff
@@ -41,42 +55,52 @@ type PlanConfig struct {
...
+	plan := &Plan{}
 	for _, p := range plans {
 		if p.ID == planID {
 			plan = p
 			break
 		}
 	}
 
+	if plan.ID == "" {
+		return nil, NewError(fmt.Sprintf("No plan with ID %s found under team ID %s and composite ID %s.", planID, teamID, compositeID))
```
Comment (pauldallen) [inline client/plan.go]: externalize error message

File: client/plan.go [hunk 9]
```diff
@@ -103,18 +129,23 @@ func (c *Client) DeletePlanByID(ctx context.Context, teamID, compositeID, planID
 		}
 	}
 
+	if !planFound {
+		return NewError(fmt.Sprintf("Failed to delete plan ID %s found under team ID %s and composite ID %s.", planID, teamID, compositeID))
```
Comment (pauldallen) [inline client/plan.go]: externalize error message

File: client/plan.go [hunk 10]
```diff
@@ -137,18 +168,23 @@ func (c *Client) EnablePlan(ctx context.Context, teamID, compositeID, planID str
 		}
 	}
 
+	if plan.ID == "" {
+		return nil, NewError(fmt.Sprintf("Failed to enable plan with ID %s found under team ID %s and composite ID %s.", planID, teamID, compositeID))
```
Comment (pauldallen) [inline client/plan.go]: externalize error message

File: client/plan.go [hunk 11]
```diff
@@ -171,6 +207,10 @@ func (c *Client) DisablePlan(ctx context.Context, teamID, compositeID, planID st
 		}
 	}
 
+	if plan.ID == "" {
+		return nil, NewError(fmt.Sprintf("Failed to disable plan with ID %s found under team ID %s and composite ID %s.", planID, teamID, compositeID))
```
Comment (pauldallen) [inline client/plan.go]: externalize error message

File: client/token.go [hunk 12]
```diff
@@ -18,52 +31,62 @@ type Token struct {
...
+	err = c.Do(ctx, "GET", tokenLink.Href, nil, &tokens)
 	if err != nil {
-		return t, err
+		return nil, err
 	}
 
-	return t, nil
+	if len(tokens) == 0 {
+		if err != nil {
+			return nil, NewError(fmt.Sprintf("No tokens found. To create a token use `coreo token add [flags]` command."))
```
Comment (pauldallen) [inline client/token.go]: externalize error message

File: client/token.go [hunk 13]
```diff
@@ -18,52 +31,62 @@ type Token struct {
...
+
 	for _, t := range tokens {
 		if t.ID == tokenID {
 			token = t
 			break
 		}
 	}
 
+	if token.ID == "" {
+		return nil, NewError(fmt.Sprintf("No token with ID %s found.", tokenID))
```
Comment (pauldallen) [inline client/token.go]: externalize error message

File: client/token.go [hunk 14]
```diff
@@ -74,13 +97,17 @@ func (c *Client) CreateToken(ctx context.Context, name, description string) (Tok
...
-	var jsonStr = []byte(tokenPlayLoad)
+	tokenPayLoad := fmt.Sprintf(`{"description":"%s","name":"%s"}`, description, name)
+	var jsonStr = []byte(tokenPayLoad)
 	err = c.Do(ctx, "POST", tokenLink.Href, bytes.NewBuffer(jsonStr), &token)
 	if err != nil {
 		return token, err
 	}
 
+	if token.ID == "" {
+		return nil, NewError(fmt.Sprintf("Failed to create token."))
```
Comment (pauldallen) [inline client/token.go]: externalize error message

File: client/token.go [hunk 15]
```diff
@@ -108,5 +137,9 @@ func (c *Client) DeleteTokenByID(ctx context.Context, tokenID string) error {
 		}
 	}
 
+	if !tokenFound {
+		return NewError(fmt.Sprintf("Failed to delete token with ID %s.", tokenID))
```
Comment (pauldallen) [inline client/token.go]: externalize error message

File: client/cloudAccount.go (hunk 3 above)
Comment (coolguru) [inline client/cloudAccount.go]: Resolved, new iteration has changes to address this.

File: client/cloudAccount.go (hunk 4 above)
Comment (coolguru) [inline client/cloudAccount.go]: Resolved, new iteration has changes to address this.

File: client/composite.go (hunk 5 above)
Comment (coolguru) [inline client/composite.go]: Resolved, new iteration has changes to address this.

File: client/gitKey.go (hunk 6 above)
Comment (coolguru) [inline client/gitKey.go]: Resolved, new iteration has changes to address this.

File: client/plan.go (hunk 7 above)
Comment (coolguru) [inline client/plan.go]: Resolved, new iteration has changes to address this.

File: client/plan.go (hunk 8 above)
Comment (coolguru) [inline client/plan.go]: Resolved, new iteration has changes to address this.

File: client/plan.go (hunk 9 above)
Comment (coolguru) [inline client/plan.go]: Resolved, new iteration has changes to address this.

File: client/plan.go (hunk 10 above)
Comment (coolguru) [inline client/plan.go]: Resolved, new iteration has changes to address this.

File: client/token.go (hunk 12 above)
Comment (coolguru) [inline client/token.go]: Resolved, new iteration has changes to address this.

File: client/token.go (hunk 13 above)
Comment (coolguru) [inline client/token.go]: Resolved, new iteration has changes to address this.

File: client/token.go (hunk 14 above)
Comment (coolguru) [inline client/token.go]: Resolved, new iteration has changes to address this.

File: client/token.go (hunk 15 above)
Comment (coolguru) [inline client/token.go]: Resolved, new iteration has changes to address this.

**Pull request merged**